app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,110 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
+  before_action :load_and_authorize_posts, except: [:create, :reply]
```

Comment:
I think it should be done in subclass. Because abstract class does not know the details about how to authorize the topic.